flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/DefaultSchedulerFactory.java

```diff
@@ -77,6 +80,7 @@ public SchedulerNG createInstance(
 				jobMasterConfiguration,
 				jobGraph.isCheckpointingEnabled())
 			.create();
+		LOG.info("Using back off time strategy {} for {} ({}).", restartBackoffTimeStrategy, jobGraph.getName(), jobGraph.getJobID());
```

Comment:
I think you mean `Using restart [...]`. I will fix it.

Reply:
I introduced a log message here because it is closest to the object creation. Moreover, I prefer to [only have field assignments](https://www.yegor256.com/2015/05/07/ctors-must-be-code-free.html) in the constructor (note however, that this already does not hold in `DefaultScheduler`). Imo, another reasonable place to log the _scheduling configuration_ is in `startSchedulingInternal()`.

Reply:
Another observation: The factory interface for the restart back off time strategies isn't really needed at the moment.

Reply:
> I introduced a log message here because it is closest to the object creation. Moreover, I prefer to [only have field assignments](https://www.yegor256.com/2015/05/07/ctors-must-be-code-free.html) in the constructor (note however, that this already does not hold in `DefaultScheduler`). Imo, another reasonable place to log the _scheduling configuration_ is in `startSchedulingInternal()`.

fine. In my understanding logging is usually not considered to be real logics (though it does something), otherwise in some cases we might find it hard to print instance internal states for trouble shooting. 
Anyway, I think it's also fine to keep the logging here.

Reply:
> Another observation: The factory interface for the restart back off time strategies isn't really needed at the moment because after `createRestartBackoffTimeStrategyFactory` we immediately call `create()`.

That's true. The `RestartBackoffTimeStrategy` creation does not need runtime information at the moment. Thought we can keep it as is before it really causes some troubles.